test-files/initVaultDev.sh

```diff
@@ -15,37 +31,49 @@ case "$(uname -s)" in
-# wait for vault server to be ready
+# Export vars that we need
+export VAULT_ADDR VAULT_TOKEN
+
+${VAULT_BIN} server -dev -dev-root-token-id ${VAULT_TOKEN} > "${VAULT_LOG}" &
+# Save the pid so we can stop it later
+echo $! > "${VAULT_PID}"
+# Wait for vault server to be ready
 sleep 5
-# create token
-./vault token create -period=10s -id="my-renewable-token" >> /tmp/vaultdev.log
-
-# create KVs
-./vault secrets enable -path=kv_v1/path/ kv >> /tmp/vaultdev.log
-./vault secrets enable -path=kv_v2/path/ kv >> /tmp/vaultdev.log
-./vault kv enable-versioning kv_v2/path/ >> /tmp/vaultdev.log
-
-# create secrets
-./vault kv put kv_v1/path/my-secret my-v1-secret=my-v1-secret-value >> /tmp/vaultdev.log
-./vault kv put kv_v2/path/my-secret my-first-secret=my-first-secret-value my-second-secret=my-second-secret-value >> /tmp/vaultdev.log
-./vault kv put kv_v2/path/json-secret @./test-files/secret.json >> /tmp/vaultdev.log
-./vault kv put kv_v1/path/json-secret @./test-files/secret.json >> /tmp/vaultdev.log
-
-# create policy
-./vault policy write VaultDevAdmin test-files/VaultPolicy.hcl >> /tmp/vaultdev.log
-./vault policy write VaultNoKV test-files/NoKVVaultPolicy.hcl >> /tmp/vaultdev.log
-# create approle
-./vault auth enable approle >> /tmp/vaultdev.log
-#./vault write auth/approle/role/my-role policies=VaultDevAdmin secret_id_ttl=100m token_num_uses=100 token_ttl=100m token_max_ttl=300m secret_id_num_uses=40 >> /tmp/vaultdev.log
-./vault write auth/approle/role/my-role policies=VaultDevAdmin token_num_uses=100 token_ttl=10s token_max_ttl=300m secret_id_num_uses=40 >> /tmp/vaultdev.log
-./vault write auth/approle/role/no-kv policies=VaultNoKV token_num_uses=2 token_ttl=30m token_max_ttl=300m secret_id_num_uses=40 >> /tmp/vaultdev.log
-
-unset VAULT_TOKEN
+
+{
+  # Create token
+  ${VAULT_BIN} token create -period=10s -id="my-renewable-token"
+
+  # Create KVs
+  ${VAULT_BIN} secrets enable -path=kv_v1/path/ kv
+  ${VAULT_BIN} secrets enable -path=kv_v2/path/ kv
+  ${VAULT_BIN} kv enable-versioning kv_v2/path/
+
+  # Create secrets
+  ${VAULT_BIN} kv put kv_v1/path/my-secret my-v1-secret=my-v1-secret-value
+  ${VAULT_BIN} kv put kv_v2/path/my-secret my-first-secret=my-first-secret-value my-second-secret=my-second-secret-value
+  ${VAULT_BIN} kv put kv_v2/path/json-secret @"${SCRIPT_DIR}/secret.json"
+  ${VAULT_BIN} kv put kv_v1/path/json-secret @"${SCRIPT_DIR}/secret.json"
+
+  # Create policies
+  ${VAULT_BIN} policy write VaultDevAdmin "${SCRIPT_DIR}/VaultPolicy.hcl"
+  ${VAULT_BIN} policy write VaultNoKV "${SCRIPT_DIR}/NoKVVaultPolicy.hcl"
+  # Create AppRoles
+  ${VAULT_BIN} auth enable approle
+  ${VAULT_BIN} write auth/approle/role/my-role policies=VaultDevAdmin token_num_uses=100 token_ttl=10s token_max_ttl=300m secret_id_num_uses=40
+  ${VAULT_BIN} write auth/approle/role/no-kv policies=VaultNoKV token_num_uses=2 token_ttl=30m token_max_ttl=300m secret_id_num_uses=40
+} >> "${VAULT_LOG}"
+
+#unset VAULT_TOKEN
```

Comment:
I see that we're removing the token in the clean up script and believe we can remove this.

Reply:
I commented it for testing before adding the cleanup script and forgot to remove it. Done!